.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,93 @@
+name: Build, test and push a Docker image
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v20[0-9][0-9].[01][0-9].[0-3][0-9]  # e.g. v2023.12.04
+      - v20[0-9][0-9].[01][0-9].[0-3][0-9]-[0-9]  # e.g. v2023.12.04-2
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    permissions:
+      contents: read
+      deployments: write
+      id-token: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get info
+        run: |
+          uname -v
+          docker info
+      - name: Create version.json
+        run: |
+          # create a version.json per
+          # https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+          printf '{"commit":"%s","version":"%s","source":"%s","build":"%s"}\n' \
+          "$GITHUB_SHA" \
+          "$GITHUB_REF_NAME" \
+          "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY" \
+          "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" > version.json
+      - name: Output version.json
+        run: cat version.json
+      - name: Build Docker images
+        run: make build
```

Comment:
in antenna's circleci we log in to docker before this step, but in eliot we don't.

I'm not sure which pattern to follow here.

Reply:
Antenna/AWS/CircleCI/DockerHub pushes images to Docker Hub.

For Antenna/GCP/GHA/GAR, we don't need to push images to Docker Hub, so I don't think we need to log into Docker Hub. That was the reasoning for not logging into Docker Hub with Eliot.

Reply:
cool, we'll continue with that then, and add docker login for rate limiting only if it becomes an issue